prime-router/src/main/kotlin/Report.kt

```diff
@@ -1608,13 +1582,24 @@ class Report : Logging {
             header: WorkflowEngine.Header,
             metadata: Metadata? = null,
         ): String {
-            return if (header.reportFile.bodyUrl != null) {
-                BlobAccess.BlobInfo.getBlobFilename(header.reportFile.bodyUrl)
+            return if (header.receiver?.topic?.isSendOriginal == true) {
+//                the externalName of the root report should equal the submission payload name parameter
+                ReportService().getRootReport(header.reportFile.reportId).externalName ?: formExternalFilename(
```

Comment:
Nit: maybe pass this in?  Seems a little weird to instantiate it here.

Reply:
My thought was that it's only needed if `isSendOriginal == true` so it was better here. What do you think?

Reply:
moved the getRootReport() call